public/src/pages/views/VisibilityView.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from "react";
+import { Enrollment } from '../../../proto/ag_pb';
+import { BootstrapButton, BootstrapClass, DynamicTable, Search } from "../../components";
+import { ILink } from '../../managers/NavigationManager';
+import { searchForCourses, sortCoursesByVisibility } from '../../componentHelper';
+
+interface VisibilityViewProps {
+    enrollments: Enrollment[];
+    onChangeClick: (enrol: Enrollment) => Promise<boolean>;
+}
+
+interface VisibilityViewState {
+    sortedCourses: Enrollment[];
+    editing: boolean;
+}
+
+export class CourseVisibilityView extends React.Component<VisibilityViewProps, VisibilityViewState> {
```

Comment:
What about `CourseListView` and similar renaming elsewhere

Reply:
Yeah, sounds reasonable. Fixing it.